framework/include/multiapps/MultiApp.h

```diff
@@ -131,6 +131,12 @@ class MultiApp : public MooseObject,
    */
   void setupPositions();
 
+  /**
+   * Create the i-th local app
+   * @param[in] i local app index
+   */
+  virtual void createLocalApp(const unsigned int & i);
```

Comment:
```suggestion
  virtual void createLocalApp(const unsigned int i);
```
It's usually faster to copy builtins

Reply:
Roy had a good comment on reference vs value
https://github.com/idaholab/moose/pull/21904#discussion_r1172738030